python/packages/ai/teams/ai/augmentations/tools_augmentation.py

```diff
@@ -0,0 +1,198 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from __future__ import annotations
+
+import json
+from typing import Dict, List, Optional, Union, cast
+
+from botbuilder.core import TurnContext
+from openai.types import chat
+
+from ...state import MemoryBase
+from ..models.chat_completion_action import ChatCompletionAction
+from ..models.prompt_response import PromptResponse
+from ..planners.plan import Plan, PredictedDoCommand, PredictedSayCommand
+from ..prompts.message import Message
+from ..prompts.sections.prompt_section import PromptSection
+from ..tokenizers.tokenizer import Tokenizer
+from ..validators.validation import Validation
+from .augmentation import Augmentation
+from .tools_constants import (
+    SUBMIT_TOOL_HISTORY,
+    SUBMIT_TOOL_OUTPUTS_MAP,
+    SUBMIT_TOOL_OUTPUTS_MESSAGES,
+    SUBMIT_TOOL_OUTPUTS_VARIABLE,
+)
+
+
+class ToolsAugmentation(Augmentation[Union[str, List[chat.ChatCompletionMessageToolCall]]]):
+    """
+    A server-side 'tools' augmentation.
+    """
+
+    _actions: List[ChatCompletionAction] = []
+
+    def __init__(self, actions: Optional[List[ChatCompletionAction]] = None) -> None:
+        if actions:
+            self._actions = actions
+
+    def create_prompt_section(self) -> Union[PromptSection, None]:
```

Comment:
this turn type should be `Optional[PromptSection]`

Reply:
`createPromptSection`/ `create_prompt_section` in the other augmentations matches the pattern of returning `PromptSection` or `undefined`, so I think it's okay as-is. Also this method isn't performing/returning anything, like in `DefaultAugmentation`/`default_augmentation`

Reply:
`Optional[T]` is the same as `Union[T, None]` https://docs.python.org/3/library/typing.html#typing.Optional

we should also update the augmentation abstract class to reflect that https://github.com/microsoft/teams-ai/blob/main/python/packages/ai/teams/ai/augmentations/augmentation.py#L30

Reply:
My bad - wasn't think of this as a python-specific nit. Thanks for clarifying.